cloud/amazon/ec2_cert

```diff
@@ -0,0 +1,219 @@
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_cert
+short_description: Administrate IAM server certificates
+description:
+    - List, upload and delete IAM server certificates for use with ELB.
+version_added: "1.9"
+options:
+  aws_secret_key:
+    description:
+      - AWS secret key. If not set then the value of the AWS_SECRET_KEY environment variable is used.
+    required: false
+    default: None
+    aliases: ['ec2_secret_key', 'secret_key' ]
+  aws_access_key:
+    description:
+      - AWS access key. If not set then the value of the AWS_ACCESS_KEY environment variable is used.
+    required: false
+    default: None
+    aliases: ['ec2_access_key', 'access_key' ]
+  cert_chain:
+    description:
+      - The contents of the certificate chain. This is typically a concatenation of the PEM-encoded public key certificates of the chain.  Required if state='present'.
+    required: false
+  cert_name:
+    description:
+      - The name for the server certificate. Do not include the path in this value. Required if state='present' or state='absent'
+    required: false
+  cert_path:
+    description:
+      - The path for the server certificate.
+    required: false
+    default: "/"
+  cert_public_key:
+    description:
+      - The contents of the public key certificate in PEM-encoded format. Required if state='present'.
+    required: false
+  cert_private_key:
+    description:
+      - The contents of the private key in PEM-encoded format. Required if state='present'.
+    required: false
+  region:
+    description:
+      - The AWS region to use. If not specified then the value of the EC2_REGION environment variable, if any, is used.
+    required: true
+    aliases: ['aws_region', 'ec2_region']
+  state:
+    description:
+      - If 'list', returns a list of all server certificates, if 'present' idempotently uploads the given certificate, and if 'absent' removes it.
+    required: false
+    default: 'present'
+'''
+
+EXAMPLES = '''
+- local_action:
+    module: ec2_cert
+    cert_name: my-cert
+    cert_path: "/my-folder/"
+    cert_public_key: "XXXX"
+    cert_private_key: "XXXX"
+    state: present
+  register: cert
+
+- debug:
+    msg: "{{ cert }}"
+'''
+
+import sys
+import time
+
+try:
+    import boto.iam
+except ImportError:
+    print "failed=True msg='boto required for this module'"
+    sys.exit(1)
```

Comment:
Duplicate PR of https://github.com/ansible/ansible-modules-core/pull/918

I'm up for ppl comparing and choosing one ove the other.

On Thursday, June 18, 2015, Bruce Pennypacker <notifications@github.com>
wrote:

> In cloud/amazon/ec2_cert
> <https://github.com/ansible/ansible-modules-extras/pull/302#discussion_r32796592>
> :
>
> > +    cert_private_key: "XXXX"
> > +    state: present
> > +  register: cert
> > +
> > +- debug:
> > +    msg: "{{ cert }}"
> > +'''
> > +
> > +import sys
> > +import time
> > +
> > +try:
> > +    import boto.iam
> > +except ImportError:
> > +    print "failed=True msg='boto required for this module'"
> > +    sys.exit(1)
>
> Rather than call sys.exit() you should call module.fail_json() if you
> can't import these modules. Here
> <https://github.com/ansible/ansible-modules-core/blob/devel/cloud/amazon/ec2.py#L507>
> is an example.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ansible/ansible-modules-extras/pull/302/files#r32796592>
> .
>